moremesecons_entity_detector/init.lua

```diff
@@ -47,7 +47,7 @@ local object_detector_scan = function (pos)
 			end
 			local isname = luaentity.name
 			for _, name in ipairs(scan_names) do
-				if isname == name or (isname == "__builtin:item" and luaentity.itemstring == name) then
+				if isname == name or (isname == "__builtin:item" and ItemStack(luaentity.itemstring):get_name() == name) then
 					return true
 				end
```

Comment:
1. These are two separate cases, thus use two separate conditions.
2. This condition does not have to be inside the loop. Let's make it more efficient.

Suggestion:
```Lua
local isname = luaentity.name
-- If the item is present as dropped item entity:
local bitemname = (isname == "__builtin:item" and ItemStack(luaentity.itemstring):get_name())
for _, name in ipairs(scan_names) do
	if name == isname then
		return true
	end
	if name == bitemname then
		return true
	end
end
```